src/lib/desmoContract-module.ts

```diff
@@ -180,21 +190,25 @@ export class DesmoContract {
     taskId: string;
     result: string;
   }> {
-    const myIexec = this.iexec as IExec;
     if (!this.isConnected) {
       throw new Error(
         'This method requires the wallet signer to be already signed-in!',
       );
     }
-    
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const self = this;
-    return new Promise( (resolve, reject) => {
-      (function loop(){
+    return new Promise((resolve, reject) => {
+      (function loop(): void {
         setTimeout(async () => {
+          if (self.iexec === undefined) {
+            reject('A connection to iExec is required!');
+            return;
+          }

```

Comment:
Thinking again about this you should throw outside the promise, immediately without waiting for the timeout function

Reply:
I put it there so that I didn't need to use the ! operator in the following lines.

For example, in `const taskDetail = await self.iexec!.task.show(taskId);`